packages/deployments/contracts/contracts/lib/Connext/ConnextUtils.sol

```diff
@@ -61,27 +335,325 @@ library ConnextUtils {
+    RelayerFeeRouter _relayerFeeRouter,
+    mapping(bytes32 => address) storage _transferRelayer
+  ) external {
+    // Ensure the relayer can claim all transfers specified
+    for (uint256 i; i < _transferIds.length; ) {
+      if (_transferRelayer[_transferIds[i]] != msg.sender)
+        revert ConnextUtils__initiateClaim_notRelayer(_transferIds[i]);
+      unchecked {
+        i++;
+      }
+    }
+
+    // Send transferIds via nomad
+    _relayerFeeRouter.send(_domain, _recipient, _transferIds);
+
+    emit InitiatedClaim(_domain, _recipient, msg.sender, _transferIds);
+  }
+
+  /**
+   * @notice Pays out a relayer for the given fees
+   * @dev Called by the RelayerFeeRouter.handle message. The validity of the transferIds is
+   * asserted before dispatching the message.
+   * @param _recipient - address on origin chain to send claimed funds to
+   * @param _transferIds - transferIds to claim
+   * @param _relayerFees - Mapping of transactionIds to fee
+   */
+  function claim(
+    address _recipient,
+    bytes32[] calldata _transferIds,
+    mapping(bytes32 => uint256) storage _relayerFees
   ) external {
-    // Approve the bridge router
-    SafeERC20Upgradeable.safeIncreaseAllowance(IERC20Upgradeable(_local), address(_bridgeRouter), _amount);
+    // Tally amounts owed
+    uint256 total;
+    for (uint256 i; i < _transferIds.length; ) {
+      total += _relayerFees[_transferIds[i]];
+      _relayerFees[_transferIds[i]] = 0;
+      unchecked {
+        i++;
+      }
+    }
 
-    _bridgeRouter.send(_local, _amount, _destination, TypeCasts.addressToBytes32(_recipient), true, _id);
+    AddressUpgradeable.sendValue(payable(_recipient), total);
+
+    emit Claimed(_recipient, total, _transferIds);
+  }
+
+  /**
+   * @notice Anyone can call this function on the origin domain to increase the relayer fee for a transfer.
+   * @param _transferId - The unique identifier of the crosschain transaction
+   */
+  function bumpTransfer(bytes32 _transferId, mapping(bytes32 => uint256) storage relayerFees) external {
+    if (msg.value == 0) revert ConnextUtils__bumpTransfer_valueIsZero();
+
+    relayerFees[_transferId] += msg.value;
+
+    emit TransferRelayerFeesUpdated(_transferId, relayerFees[_transferId], msg.sender);
+  }
+
+  // ============ Private Functions ============
+
+  /**
+   * @notice Performs some sanity checks for `execute`
+   * @dev Need this to prevent stack too deep
+   */
+  function _executeSanityChecks(
+    ExecuteLibArgs calldata _args,
+    mapping(bytes32 => address) storage _transferRelayer,
+    mapping(bytes32 => bool) storage _reconciledTransfers
+  ) private returns (bytes32, bool) {
+    if (_args.executeArgs.routers.length > _args.maxRoutersPerTransfer)
+      revert ConnextUtils__execute_maxRoutersExceeded();
+
+    bytes32 transferId = _getTransferId(_args);
+
+    // require this transfer has not already been executed
+    if (_transferRelayer[transferId] != address(0)) {
```

Comment:
shouldn't this be `_transferRelayer[transferId] == address(0)` ?

Reply:
no, `_transferRelayer[transferId] == address(0)` means it hasn't been executed thus it doesn't rever